graf2d/fitsio/src/TFITS.cxx

```diff
@@ -509,12 +517,79 @@ Bool_t TFITSHDU::LoadHDU(TString& filepath_filter)
                         for (long component = 0; component < repeat; component++) {
                            vec[component] = array[offset++];
                         }
-                        fCells[cellindex++].fRealVector = vec;
+                        fCells[cellindex++].fRealArray = vec;
                      }
                      delete[] array;
                   }
                }
             }
+         } else if (typecode < 1) {
+            // this column contains variable-length arrays
+            fColumnsInfo[colnum].fType = kRealVector;
+
+            // null variables needed by the fits_read_col
+            double nulval = 0;
+            int anynul=0;
+            
+            // loop over rows to derive the total memory requirement
+            fColumnsInfo[colnum].fRowStart.assign(table_rows+1, 0);
+            fColumnsInfo[colnum].fVarLengths.assign(table_rows, 0);
+            fColumnsInfo[colnum].fRowStart[0] = 0;
+
+            for (long row = 0; row < table_rows; row++) {
+               long offset = 0;
+               long repeat = 0;
```

Comment:
This variable name (see CI warnings) is already used in prior code.   Either reduce the 'lifetime/scope' of the previous variable or rename this one to avoid (future developer reader the code) confusion.  thanks.

Reply:
Also done!